include/oneapi/dpl/pstl/parallel_backend_tbb.h

```diff
@@ -34,7 +34,27 @@
 #include <tbb/parallel_invoke.h>
 #include <tbb/task_arena.h>
 #include <tbb/tbb_allocator.h>
+
+#if _WIN32 || _WIN64
```

Comment:
May be better to move this block before `tbb` includes?

Reply:
In my opinion, the current approach is best since it targets just the problematic header and limits where we use this macro to only where it is needed.

Reply:
I can see the point about wrapping all tbb headers, as we dont know if another TBB header might include `tbb/enumerable_thread_specific.h`.  

However, I see the requirement for this define as undesirable, so I'd rather limit it in our code by version to allow TBB to handle it as they see fit in the future. 

I'm OK with whatever the consensus is here, though.